src/st/no0/e_ouija_table.c

```diff
@@ -0,0 +1,317 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+#include "no0.h"
+
+#define VASE self + 1
+#define BEAKER self + 2
+#define WALKING_TABLE self + 3
+
+// Main
+extern u16 D_us_80180C14[];
+extern s16 D_us_801825CC[];
+extern u16 D_us_801825D4[];
+extern u8 D_us_801825F4[];
+
+// Components
+extern u16 D_us_80180C20[];
+extern u16 D_us_801825E4[];
+
+typedef enum {
+    /* 0 */ OUIJA_TABLE_INIT,
+    /* 1 */ OUIJA_TABLE_INIT_SUBENTITIES,
+    /* 2 */ OUIJA_TABLE_COME_ALIVE,
+    /* 3 */ OUIJA_TABLE_ATTACK,
+    /* 4 */ OUIJA_TABLE_UNUSED,
+    /* 255 */ OUIJA_TABLE_DEBUG = 255,
+} OuijaTableSteps;
+
+typedef enum {
+    /* 0 */ OUIJA_COMPONENT_INIT,
+    /* 1 */ OUIJA_COMPONENT_CHECK_SPAWN,
+    /* 2 */ OUIJA_COMPONENT_INIT_POS,
+    /* 3 */ OUIJA_COMPONENT_CIRCLE_TABLE,
+    /* 4 */ OUIJA_COMPONENT_SEEK_ATTACK,
+    /* 8 */ OUIJA_COMPONENT_DEATH = 8,
+    /* 16 */ OUIJA_COMPONENT_NOOP = 16
+} OuijaTableComponentSteps;
+
+typedef enum {
+    /* 0 */ OUIJA_COMPONENT_BEGIN_DEATH,
+    /* 1 */ OUIJA_COMPONENT_FALL_TO_FLOOR,
+    /* 2 */ OUIJA_COMPONENT_DIE_IN_FLAMES
+} OuijaTableComponentDeathSubSteps;
+
+// Main Ouija Table entity
+void EntityOuijaTable(Entity* self) {
+    Entity* otherEntity;
+
+    if (self->flags & FLAG_DEAD) {
+        otherEntity = AllocEntity(&g_Entities[224], &g_Entities[256]);
+        if (otherEntity != NULL) {
+            CreateEntityFromEntity(E_EXPLOSION, self, otherEntity);
+            otherEntity->params = 1;
+        }
+        PlaySfxPositional(SFX_STUTTER_FIREBALL);
+        // Destroy the spinning objects on the table
+        otherEntity = VASE;
+        otherEntity->flags |= FLAG_UNK_2000;
+
+        otherEntity = BEAKER;
+        otherEntity->flags |= FLAG_UNK_2000;
+        DestroyEntity(self);
+        return;
+    }
+
+    switch (self->step) {
+    case OUIJA_TABLE_INIT:
+        InitializeEntity(D_us_80180C14);
+        self->animCurFrame = 5;
+        self->hitboxOffY = 2;
+        return;
+    case OUIJA_TABLE_INIT_SUBENTITIES:
+        if (UnkCollisionFunc3(D_us_801825D4) & 1) {
+            // Spawn objects on the table
+            otherEntity = VASE;
+            CreateEntityFromEntity(E_OUIJA_TABLE_COMPONENT, self, otherEntity);
+            otherEntity->params = 0;
+            otherEntity->ext.ouijaTableContents.parent = self;
+            otherEntity->posX.i.hi -= 4;
+            otherEntity->posY.i.hi -= 16;
+
+            otherEntity = BEAKER;
+            CreateEntityFromEntity(E_OUIJA_TABLE_COMPONENT, self, otherEntity);
+            otherEntity->params = 1;
+            otherEntity->ext.ouijaTableContents.parent = self;
```

Comment:
I didn't check but you're going to want to have a static_assert on the pointer and an item afterwards to make sure it all aligns

Reply:
Sorry not sure what you mean by an item afterwards, but I added a static assert, hopefully that's right

Reply:
Failing on the i386 build now. I'm not sure how to fix sorry, if someone can run me through what I need to change that would be great